kitsune/src/http/pagination.rs

```diff
@@ -26,10 +26,46 @@ where
             .collect::<Vec<String>>()
             .join(", ");
 
-        // as long as we pass valid links this should never panic
-        res.headers_mut()
-            .insert("Link", HeaderValue::from_str(&value).unwrap());
+        // TODO: log an error when HeaderValue conversion fails
+        if let Ok(header_value) = HeaderValue::from_str(&value) {
+            res.headers_mut().insert("Link", header_value);
+        }
 
         Ok(res)
     }
 }
+
+pub fn new_link_header<I, D: Display, F: Fn(&I) -> D>(
+    collection: &Vec<I>,
+    limit: usize,
+    base_url: &str,
+    get_key: F,
+) -> Option<Link<Vec<(&'static str, String)>>> {
+    if collection.is_empty() {
+        None
+    } else {
+        let next = (
+            "next",
+            format!(
+                "{}/api/v1/timelines/public?limit={}&max_id={}",
```

Comment:
But this isn't quite right for the `follow_requests` route then, no?

Something that one could do is pass in the URI from the handler.  
So you don't have to type it out manually, you could use something like the `OriginalUri` extractor in the handlers.

Reply:
The only concern I have with the approach of passing in the URI is that it's yet *another* parameter and that makes the list of parameters pretty unwieldy, to say the least

Reply:
Not changing the route was a big brainfart on my part, sorry, but now I changed it to pass the path as an argument

> The only concern I have with the approach of passing in the URI is that it's yet _another_ parameter and that makes the list of parameters pretty unwieldy, to say the least

That's why I was pretty hesitant at first to make a dedicated function for that, but still, a function with 5 arguments looks better that similar 20 lines of code. 

The only issue now is that by adding `OriginalUri` to the `/api/v1/accounts/statuses` route, there are 8 arguments now which is apparently too much for clippy

Reply:
I just checked the function signature. It looks like we have a duplicated extraction with `auth_data` and `user_data`.  
We should remove one of those anyway. I don't really care particularly about which one we keep, feel free to just choose one.

Reply:
Oh right lol, I didn't even notice, but I removed one of these now so it should be ok